contracts/RegistryAdapters/IronBankRegistryAdapter.sol

```diff
@@ -366,6 +373,58 @@ contract RegistryAdapterIronBank is Ownable {
             });
     }
 
+    /**
+     * Internal method for getting supply balance data
+     */
+    function _userSupplyBalance(address accountAddress, ICyToken asset, address tokenAddress, uint256 tokenPriceUsdc)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 supplyBalanceShares = asset.balanceOf(accountAddress);
+        uint256 supplyBalanceUnderlying =
+            (supplyBalanceShares * asset.exchangeRateStored()) / 10**18;
+        return IOracle(oracleAddress).getNormalizedValueUsdc(
+            tokenAddress,
+            supplyBalanceUnderlying,
+            tokenPriceUsdc
+        );
+    }
+
+    /**
+     * Internal method for getting borrow balance data
+     */
+    function _userBorrowBalance(address accountAddress, ICyToken asset, address tokenAddress, uint256 tokenPriceUsdc)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 borrowBalanceShares = asset.borrowBalanceStored(accountAddress);
+        return IOracle(oracleAddress).getNormalizedValueUsdc(
+            tokenAddress,
+            borrowBalanceShares,
+            tokenPriceUsdc
+        );
+    }
+
+    /**
+     * Internal method for getting collateral balance data
+     */
+    function _userCollateralBalance(address accountAddress, ICyToken asset, address tokenAddress, uint256 tokenPriceUsdc)
+        internal
+        view
+        returns (uint256)
+    {
+        (, uint256 collateralBalanceShare, , ) = asset.getAccountSnapshot(accountAddress);
```

Comment:
Does `getAccountSnapshot` exist on version 0 if cyTokens?

Reply:
Yes, function `getAccountSnapshot` exists in the very beginning.